File: python_bot/games/anagrams.py
#!/usr/bin/env python3
"""لعبة الحروف - رتب الحروف المشوشرة"""
import random
from games.base import BaseGame

class AnagramsGame(BaseGame):
    def get_game_id(self):
        return "anagrams"
    def scramble(self, word):
        chars = list(word)
        random.shuffle(chars)
        scrambled = ''.join(chars)
        if scrambled == word:
            chars[0], chars[-1] = chars[-1], chars[0]
            scrambled = ''.join(chars)
        return scrambled
    
    def generate_question(self, session):
        words = ["كتاب", "مدرسة", "جامعة", "حديقة", "مطبخ", "نافذة", "ساعة", "قلم", "وردة", "شمس",
                 "باب", "سيارة", "طائرة", "جبل", "نهر", "غابة", "محطة", "مكتب", "كرسي", "هاتف"]
        word = random.choice(words)
        scrambled = self.scramble(word)
        return {"question": scrambled, "correct_answer": word}
    def check_answer(self, session, user_id, answer):
        q = session['question_data']
        return answer.strip().lower() == q['correct_answer'].strip().lower()
    def render_question(self, session, question):
        return f"🔀 الحروف المشوشة: {question['question']}\n\nرتبها لتكوين كلمة صحيحة:"

File: src/games/individual/anagrams.ts
import { GameSession, AiResponse } from '../../types';
import { BaseGame } from '../base';
import { AiService } from '../../ai/api';
import { SessionManager } from '../../session/manager';
import { PointsSystem } from '../../points/system';
import { GroupManager } from '../../group/manager';
import { LeaderboardManager } from '../../leaderboard/manager';
import { KvStore } from '../../db/kv';
import { DbStore } from '../../db/d1';
import { TelegramApi } from '../../telegram';
import { Env } from '../../types';

export class AnagramsGame extends BaseGame {
  private words = ['كتاب', 'مدرسة', 'جامعة', 'حديقة', 'مطبخ', 'نافذة', 'ساعة', 'قلم', 'وردة', 'شمس', 'باب', 'سيارة', 'طائرة', 'جبل', 'نهر', 'غابة', 'محطة', 'مكتب', 'كرسي', 'هاتف'];

  constructor(
    ai: AiService, sessionManager: SessionManager, pointsSystem: PointsSystem,
    groupManager: GroupManager, leaderboardManager: LeaderboardManager,
    kv: KvStore, db: DbStore, telegram: TelegramApi, env: Env
  ) { super(ai, sessionManager, pointsSystem, groupManager, leaderboardManager, kv, db, telegram, env); }

  getGameId(): string { return 'anagrams'; }

  private scramble(word: string): string {
    const arr = word.split('');
    for (let i = arr.length - 1; i > 0; i--) {
      const j = Math.floor(Math.random() * (i + 1));
      [arr[i], arr[j]] = [arr[j], arr[i]];
    }
    if (arr.join('') === word) {
      [arr[0], arr[arr.length - 1]] = [arr[arr.length - 1], arr[0]];
    }
    return arr.join('');
  }

  async generateQuestion(session: GameSession, round: number): Promise<AiResponse> {
    const word = this.words[round % this.words.length];
    const scrambled = this.scramble(word);
    return {
      success: true, question: scrambled, options: [], correctAnswer: word,
      storyBranch: [], choices: [], hint: '', explanation: '',
      metadata: { original: word },
    };
  }

  async checkAnswer(session: GameSession, userId: number, answer: string): Promise<boolean> {
    const q = session.currentQuestion;
    if (!q) return false;
    return answer.trim().toLowerCase() === q.correctAnswer.trim().toLowerCase();
  }

  renderQuestion(session: GameSession, q: AiResponse): string {
    return `🔀 الحروف المشوشة: ${q.question}\n\nرتبها لتكوين كلمة صحيحة:`;
  }

  renderRoundStart(session: GameSession): string {
    return '🎯 لعبة الحروف!\n\nسيتم عرض حروف مشوشرة وعليك ترتيبها لتكوين كلمة صحيحة.';
  }
}